### PR TITLE
Header Slug Scrolling

### PR DESCRIPTION
**Pull request by erikaannesmith**
⚡ fridayfridayfriday already! 🌵 

this PR renders link icons next to each header, enabling users to generate a slug for//automatically scroll to that section.
this is the way that Victory handles this same functionality, but I'm open to other approaches if anything comes to you!

**click around. it's pretty satisfying.**
Screen Shot 2020-12-11 at 12 59 33 PM

### CODE REVIEW

File: src/components/docs/Doc.jsx [hunk 1]
```diff
@@ -1,69 +1,96 @@
...
+  level: PropTypes.number,
 };
 
 export default function Doc({ doc, toc }) {
   if (!doc || !toc) {
     return null;
   }
-  const compiled = compile(doc.content);
+
+  const renderers = { heading: renderHeading };
```
Comment (erikaannesmith) [inline src/components/docs/Doc.jsx:55]: I imagine this is where we'll add something custom to handle `react-live` clips :)

Comment (boygirl) [reply src/components/docs/Doc.jsx:55]: Yep! You should be able to check on the `language` of the code block and either render a `react-live` component or a `prism` highlighted code block

**Comment by erikaannesmith** [comment]
also to note! the reasons we need all this crazy logic are:
A) sweet scrolling animations
B) the fixed header makes us offset the content we're scrolling to, which you would think would be straightforward, but alas..

File: package.json [hunk 2]
```diff
@@ -43,11 +43,13 @@
     "version": "yarn clean && yarn build"
   },
   "dependencies": {
+    "@juggle/resize-observer": "^3.2.0",
     "formidable-oss-badges": "^0.3.5",
     "markdown-toc": "^1.2.0",
     "prism-react-renderer": "^1.1.1",
     "react-copy-to-clipboard": "^5.0.2",
     "react-live": "^2.2.2",
+    "react-scroll": "^1.8.1",
```
Comment (boygirl) [inline package.json:52]: I think `react-markdown` should be a dependency

**Review by boygirl — CHANGES_REQUESTED** [review]
weird-scroll

I'm seeing some odd scrolling behavior on the last anchor tag on a page

**Comment by erikaannesmith** [comment]
> weird-scroll
> 
> I'm seeing some odd scrolling behavior on the last anchor tag on a page

ah! I think maybe this has to do with them having the same title?! I'll add an index or something and see if that helps :) good catch!

File: package.json (hunk 2 above)
Comment (erikaannesmith) [reply package.json:52]: ah yes! I'm surprised this didn't yell at me

**Review by boygirl — APPROVED** [review]

**Comment by boygirl** [comment]
Looks good! Edge case scroll behavior is captured in https://github.com/FormidableLabs/formidable-oss-landers/issues/24

**Pull request merged**